### PR TITLE
Basic player health system

### PR DESCRIPTION
**Pull request by TheMadHatter01**

### CODE REVIEW

File: scenes/player/HealthStats.gd [hunk 1]
```diff
@@ -0,0 +1,29 @@
+extends Node
```
Comment (AWarmCupOfTea) [inline scenes/player/HealthStats.gd:1]: What's the difference between this file and HealthBar.gd ?

File: scenes/player/player.gd [hunk 2]
```diff
@@ -86,3 +87,8 @@ func get_move_input_direction() -> int:
 	if Input.is_action_pressed("move_left"):
 		move_direction += MOVE_DIRECTION.LEFT
 	return move_direction
+
+
+func _on_Area2D_area_entered(area):
```
Comment (AWarmCupOfTea) [inline scenes/player/player.gd:92]: Should this really be in the player class?

**Review by AWarmCupOfTea — CHANGES_REQUESTED** [review]

**Comment by AWarmCupOfTea** [comment]
Pleas use this convention to name commits: https://cbea.ms/git-commit/

For example, `Add basic player health system`

File: scenes/player/HealthStats.gd (hunk 1 above)
Comment (TheMadHatter01) [reply scenes/player/HealthStats.gd:1]: the HealthBar.gd was to get the health bar to work

File: scenes/player/player.gd (hunk 2 above)
Comment (TheMadHatter01) [reply scenes/player/player.gd:92]: There is probably a better way to do it, but thats how i know how to do it.

Comment (AWarmCupOfTea) [reply scenes/player/player.gd:92]: Well, any class should follow [Single-responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle), so the logic of this Area2D should be inside its class.

File: scenes/player/HealthStats.gd (hunk 1 above)
Comment (AWarmCupOfTea) [reply scenes/player/HealthStats.gd:1]: I see duplicate code in both files, and I don't understand why is it needed.

Comment (TheMadHatter01) [reply scenes/player/HealthStats.gd:1]: ok, ill try cleaning up the code

File: scenes/player/player.gd (hunk 2 above)
Comment (TheMadHatter01) [reply scenes/player/player.gd:92]: i see, so ill attach a script to the area2d with that bit of code?

Comment (AWarmCupOfTea) [reply scenes/player/player.gd:92]: Yes, you need to figure out how to keep the logic inside their relevant classes. It's a bad idea to pile everything in one place.

Comment (TheMadHatter01) [reply scenes/player/player.gd:92]: Ok, I won't be able to work on it today or tomorrow, so I'll continue on Sunday

**Pull request closed without merging**